File: 2Star AdvLvl/HP18.java
import java.util.Scanner;

public class Main {
    public static void luckyNumber() {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int a = scanner.nextInt();
        int b = scanner.nextInt();
        int[] arr = new int[n];
        int alice = 0, bob = 0, albob = 0;

        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
            if (arr[i] % a == 0 && arr[i] % b == 0) {
                albob++;
            } else if (arr[i] % a == 0) {
                bob++;
            } else if (arr[i] % b == 0) {
                alice++;
            }
        }
        if (alice == 0 && bob == 0 && albob != 0)
            System.out.println("BOB");

        else if (alice == 0 && bob == 0 && albob == 0)
            System.out.println("ALICE");

        else if (albob == 0) {
            if (alice >= bob)
                System.out.println("ALICE");
            else
                System.out.println("BOB");
        } else {
            if (alice > bob)
                System.out.println("ALICE");
            else
                System.out.println("BOB");
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        while (t-- > 0) {
            luckyNumber();
        }
    }
}

File: 2Star AdvLvl/HP18.py
def luckyNumber():
    n, a, b = map(int, input().split())
    arr = list(map(int, input().split()))
    alice = 0
    bob = 0
    albob = 0

    for num in arr:
        if num % a == 0 and num % b == 0:
            albob += 1
        elif num % a == 0:
            bob += 1
        elif num % b == 0:
            alice += 1

    if alice == 0 and bob == 0 and albob != 0:
        print("BOB")
    elif alice == 0 and bob == 0 and albob == 0:
        print("ALICE")
    elif albob == 0:
        if alice >= bob:
            print("ALICE")
        else:
            print("BOB")
    else:
        if alice > bob:
            print("ALICE")
        else:
            print("BOB")

t = int(input())
for _ in range(t):
    luckyNumber()
